### DISCUSSION TITLE
Works on Mac M1

### DISCUSSION THREAD

**Discussion by vapidvim** [volotat/SD-CN-Animation#126] [General]
The extension is working for me!

I run a1111 using these parameters in webui-user.sh and in general it is faster and stopped crashing:
export COMMANDLINE_ARGS="--skip-torch-cuda-test --upcast-sampling --opt-sub-quad-attention --use-cpu interrogate" 

I also added this to keep torch updated:
\# python3 venv without trailing slash (defaults to ${install_dir}/${clone_dir}/venv)
venv_dir="venv-torch-nightly"

\# install command for torch
export TORCH_COMMAND="pip install --pre torch torchvision -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html"

Also be sure to check "Upcast cross attention layer to float32" at Settings->Stable Diffusion

**Comment by vapidvim** [comment]
https://github.com/volotat/SD-CN-Animation/assets/5431091/f8f6becf-79fa-48db-927c-adafa81a83ea

Here is a quick example from this morning on my Mac. I lowered strengths since CN is not working.

**Comment by vapidvim** [comment]
I updated my control net extension and it is there now and working 100%. :)

**Comment by kchiefs** [comment]
I cannot get it to work on my M2 mac.  I just get the error Torch not compiled with CUDA enabled

**Reply by vapidvim** [reply]
> I cannot get it to work on my M2 mac. I just get the error Torch not compiled with CUDA enabled

Issue with the current version. Try the last build found here: https://github.com/volotat/SD-CN-Animation/tree/c3c0972c0a3a2697533f662ab2e9cb24413fbd0c

**Reply by kchiefs** [reply]
still getting the same error.

**Reply by kchiefs** [reply]
nm, just found out i suck at git.  Its running now and no error, thank you so much.

**Reply by pjpmccarthy52** [reply]
Thanks!  Using your recommended branch fixed the CUDA issue for me too on my M1 studio.
